p2p/switch_test.go

```diff
@@ -944,3 +956,70 @@ func TestReactorSpecificBroadcast(t *testing.T) {
 	assert.Equal(t, totalPeers-1, barCnt)
 	assert.Equal(t, reactorPeerLimit, limitedCnt)
 }
+
+func TestReactorSpecificPeerRemoval(t *testing.T) {
```

Comment:
does it make sense to also test removing a peer from all reactors and see if we successfuly disconnect?

Reply:
This test actually have 2 steps, and in the second peer is completely removed from all reactors and we make assertions to ensure it's removed and stopped.